Source/WebKit/UIProcess/ios/forms/WKDatePickerPopoverController.mm

```diff
@@ -96,16 +93,43 @@ - (instancetype)initWithDatePicker:(UIDatePicker *)datePicker
         [[_backgroundView trailingAnchor] constraintEqualToAnchor:self.trailingAnchor],
         [[_backgroundView topAnchor] constraintEqualToAnchor:self.topAnchor],
         [[_backgroundView bottomAnchor] constraintEqualToAnchor:self.bottomAnchor constant:-toolbarBottomMargin],
-        [[_datePicker heightAnchor] constraintEqualToConstant:datePickerSize.height],
-        [[_datePicker leadingAnchor] constraintEqualToAnchor:self.leadingAnchor],
-        [[_datePicker trailingAnchor] constraintEqualToAnchor:self.trailingAnchor],
-        [[_datePicker topAnchor] constraintEqualToAnchor:self.topAnchor],
-        [[_datePicker bottomAnchor] constraintEqualToSystemSpacingBelowAnchor:[_accessoryView topAnchor] multiplier:1],
+        [[mainView heightAnchor] constraintEqualToConstant:mainViewSize.height],
+        [[mainView leadingAnchor] constraintEqualToAnchor:self.leadingAnchor],
+        [[mainView trailingAnchor] constraintEqualToAnchor:self.trailingAnchor],
+        [[mainView topAnchor] constraintEqualToAnchor:self.topAnchor],
+        [[mainView bottomAnchor] constraintEqualToSystemSpacingBelowAnchor:[_accessoryView topAnchor] multiplier:1],
         [[_accessoryView leadingAnchor] constraintEqualToAnchor:self.leadingAnchor constant:accessoryViewHorizontalMargin],
         [[_accessoryView trailingAnchor] constraintEqualToAnchor:self.trailingAnchor constant:-accessoryViewHorizontalMargin],
         [[_accessoryView heightAnchor] constraintEqualToConstant:accessoryViewSize.height],
         [[_accessoryView bottomAnchor] constraintEqualToAnchor:[_backgroundView bottomAnchor]],
     ]];
+}
+
+- (instancetype)initWithCalendarView:(UICalendarView *)calendarView selectionWeekOfYear:(UICalendarSelectionWeekOfYear*)weekSelection
+{
+    if (!(self = [super initWithFrame:CGRectZero]))
+        return nil;
+
+    _calendarView = calendarView;
+    NSCalendar *cal = [NSCalendar calendarWithIdentifier:NSCalendarIdentifierISO8601];
```

Comment:
we usually try to avoid abbreviations; also, this should be `RetainPtr name` instead of `NSCalendar *name`

Reply:
Is that a +1 method?

Reply:
It's not a +1 method. Smart pointer guidelines would recommend using RetainPtr here though (`RetainPtr calendar = [NSCalendar calendarWithIdentifier:NSCalendarIdentifierISO8601];`)